### PR TITLE
Run line when nothing is selected

### PR DESCRIPTION
**Pull request by wch**
When no text is selected, pressing cmd-Enter runs the entire line, and advances the cursor to the next line. (This is similar to how rstudio works.)

If there are multiple regions, it only does this for those regions that have no selection (i.e., cursors with no selection).

The code for moving the cursor comes is borrowed from:
http://www.sublimetext.com/forum/viewtopic.php?f=6&t=2819

It basically removes the old regions and adds new ones.

### CODE REVIEW

File: Rtools.py [hunk 1]
```diff
@@ -34,13 +34,23 @@ def run(self, edit):
...
+            sel = self.view.substr(region)
+            # If empty selection, use whole line
+            if sel == "":
+                sel = self.view.substr(self.view.line(region))
+
+            selection += sel + "\n"
         selection = (selection[::-1].replace('\n'[::-1], '', 1))[::-1]
 
+        # For empty selections, advance cursor one line
+        self.advanceCursors()
```
Comment (jlegewie) [inline Rtools.py]: Doesn't that always advance the cursor? Should this be place under the if in line 39?

Comment (wch) [inline Rtools.py]: The function only advances cursors if there's no text selected. I'll see what I can do about putting it under line 39, though. I previously thought that you had to `clear()` all the regions and then add them all back, but I was wrong about that.

**Comment by wch** [comment]
OK, I changed the function to `advanceCursor()`, and run it for each empty region, as suggested. This resulting code is significantly simpler.

Before you merge, let me know and I'll squash the commits so that there's not extraneous stuff in the commit history. (Or you can rebase and squash them yourself.)

**Comment by jlegewie** [comment]
okay, great. I am ready to merge. Let me know when you done with cleaning....

**Comment by wch** [comment]
OK, it should be good to go. I also simplified the logic for detecting empty regions.

**Comment by jlegewie** [comment]
okay, great. Thanks!

Could you test this: https://github.com/jlegewie/SublimeQuickLook
I think you will like it! Still has a lot of problems and doesn't run smoothly but the idea is that you can access R help files with a single key (super+shift+h) stroke from within ST2. Do test it you have to copy it to your ST2 package folder and extract the R-help.zip file....

**Pull request merged**